# Heart Attack Analysis & Prediction 

<br>
<br>
<img src="images/heart-attack-patient.jpeg" alt="Drawing" style="width: 500px;"/>
<br>
<b>Author</b>: Juana Tavera

## Overview

### Business Problem
A health clinic in the Metro Atlanta area has called on Tavera Analytics to come up with model that is able to classify if a patient is at risk of having a heart attack or not based on certain features, in the hopes of reducing the workload of the staff.

### Dataset - Heart Attack Analysis & Prediction 
This dataset was found on Kaggle. Here is the direct [link](https://www.kaggle.com/datasets/rashikrahmanpritom/heart-attack-analysis-prediction-dataset).

In [4]:
#importing the necessary packages 
import pandas as pd


#reading in the data
heart_attack = pd.read_csv('data/heart.csv')

## Exploratory Data Analysis

In [6]:
display(heart_attack.head())
display(heart_attack.info())

age  sex  cp  trtbps  chol  fbs  restecg  thalachh  exng  oldpeak  slp  \
0   63    1   3     145   233    1        0       150     0      2.3    0   
1   37    1   2     130   250    0        1       187     0      3.5    0   
2   41    0   1     130   204    0        0       172     0      1.4    2   
3   56    1   1     120   236    0        1       178     0      0.8    2   
4   57    0   0     120   354    0        1       163     1      0.6    2   

   caa  thall  output  
0    0      1       1  
1    0      2       1  
2    0      2       1  
3    0      2       1  
4    0      2       1

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 303 entries, 0 to 302
Data columns (total 14 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   age       303 non-null    int64  
 1   sex       303 non-null    int64  
 2   cp        303 non-null    int64  
 3   trtbps    303 non-null    int64  
 4   chol      303 non-null    int64  
 5   fbs       303 non-null    int64  
 6   restecg   303 non-null    int64  
 7   thalachh  303 non-null    int64  
 8   exng      303 non-null    int64  
 9   oldpeak   303 non-null    float64
 10  slp       303 non-null    int64  
 11  caa       303 non-null    int64  
 12  thall     303 non-null    int64  
 13  output    303 non-null    int64  
dtypes: float64(1), int64(13)
memory usage: 33.3 KB


None

This is a very simple dataset it only has 303 entries, 14 columns. No null values which is perfect for going straight into modeling which this tutorial is all about but we still want to explore our data a bit more.

From this dataset can see that although the dtypes of the columns are all numerical there are some columns that are considered to be categorical:
- `sex`
- `cp`: chest pain 
- `fbs`: fasting blood sugar
- `restecg`: resting electromagentic results
- `exng`: exercise induced angina
- `slp`: slope
- `caa`: number of major vessels
- `thall`: thalassemia


In [15]:
#lets look at the value counts for the categorical variables
cat_heart_attack = heart_attack[['sex', 'cp', 'fbs', 'restecg', 'exng', 'slp', 'caa']]

for column, values in cat_heart_attack.iteritems():
    print(f'Column Name: {column}')
    print('Value counts: ')
    print(values.value_counts())
    print("")

Column Name: sex
Value counts: 
1    207
0     96
Name: sex, dtype: int64

Column Name: cp
Value counts: 
0    143
2     87
1     50
3     23
Name: cp, dtype: int64

Column Name: fbs
Value counts: 
0    258
1     45
Name: fbs, dtype: int64

Column Name: restecg
Value counts: 
1    152
0    147
2      4
Name: restecg, dtype: int64

Column Name: exng
Value counts: 
0    204
1     99
Name: exng, dtype: int64

Column Name: slp
Value counts: 
2    142
1    140
0     21
Name: slp, dtype: int64

Column Name: caa
Value counts: 
0    175
1     65
2     38
3     20
4      5
Name: caa, dtype: int64

